torch_geometric/nn/summary.py

```diff
@@ -0,0 +1,156 @@
+from collections import defaultdict
+from time import perf_counter
+from typing import Any, Sequence
+
+import torch
+import torch.nn as nn
+
+
+def summary(
+    model: nn.Module,
+    *inputs: Sequence[Any],
+    device: torch.device = 'cpu',
+    max_depth: int = 3,
+) -> str:
+    r"""Summarizes the given PyTorch model. Summarized information includes
+    (1) Layer names, (2) Input/output shapes,
+    (3) # of parameters, and (4) Excutation time.
+
+    .. code-block:: python
+
+        import torch
+        from torch_geometric.nn import summary
+        from torch_geometric.nn.models import GCN
+
+        model = GCN(128, 64, 2, out_channels=32)
+        x = torch.randn(100, 128)
+        edge_index = torch.randint(100, size=(2, 20))
+        print(summary(model, x, edge_index))
+
+    Args:
+        model (nn.Module): PyTorch model to summarize.
+        inputs (Sequence[Any]): Arguments for the model's
+            forward pass.
+        device (torch.device): Device for model and input_data.
+            (default: :obj:`"cpu"`)
```

Comment:
Do we desperately need this argument? If the model is on GPU, inputs should be on GPU as well.

Reply:
Agreed.